python/whylogs/api/pyspark/experimental/profiler.py

```diff
@@ -75,6 +77,51 @@ def collect_column_profile_views(
     return collected_profile_views
 
 
+def _collect_column_profile_views_batched(
+    batch_size: int, input_df: SparkDataFrame, schema: Optional[DatasetSchema] = None
+) -> Dict[str, ColumnProfileView]:
+    if batch_size <= 0:
+        logger.warning(
+            f"Batched pyspark column processing was called with batch_size of {batch_size}, falling back to non-batched profiling."
+        )
+        return collect_column_profile_views(input_df=input_df, schema=schema)
+
+    columns = input_df.columns
+    num_batches = len(columns) // batch_size  # Total number of batches
+
+    # Initialize an empty dataset_profile
+    combined_column_views = dict()
+
+    # process the column profiles in batches
+    for batch_num in range(num_batches):
+        start_idx = batch_num * batch_size
+        end_idx = (batch_num + 1) * batch_size
+        selected_columns = columns[start_idx:end_idx]
+        column_partitioned_batch_df = input_df.select(*selected_columns)
+        column_views_dict = collect_column_profile_views(input_df=column_partitioned_batch_df, schema=schema)
+        combined_column_views.update(column_views_dict)
+
+    # Handle the last batch (if any) that may have fewer columns
+    last_batch_columns = columns[num_batches * batch_size :]
+    if last_batch_columns:
+        column_partitioned_batch_df = input_df.select(*last_batch_columns)
+        # Process the last batch DataFrame
+        column_views_dict = collect_column_profile_views(input_df=column_partitioned_batch_df, schema=schema)
+        combined_column_views.update(column_views_dict)
+
+    return combined_column_views
+
+
+def _get_column_batch_size(schema: Optional[DatasetSchema] = None) -> Optional[int]:
+    if schema and schema.default_configs:
+        batch_size = schema.default_configs.column_batch_size
+    else:
+        default_config = MetricConfig()
```

Comment:
This doesn't make sense to me as part of the `MetricConfig`. Maybe a parameter on the schema?

Reply:
moved out of MetricConfig